fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContextConfig.java

```diff
@@ -597,6 +628,39 @@ public Builder setRecordContextProperties(@Nonnull final RecordLayerPropertyStor
             return this;
         }
 
+        /**
+         * Get tags used for throttling.
+         * @return throttling tags
+         */
+        @Nonnull
+        public Set<String> getTags() {
+            return tags;
+        }
+
+        /**
+         * Set tags used for throttling.
+         * @param tags new set of tags
+         */
+        public void setTags(@Nonnull final Set<String> tags) {
+            this.tags = tags;
```

Comment:
I guess I was a little surprised that the builder here took a `Set<String>` instead of having an `addTag` method. As written, it looks like this has the advantage that it doesn't need to allocate any objects if the (default) empty set is used, which nice, so maybe this is preferable.

Reading the docs, it also sounds like each transaction can have at most 5 tags, and each tag should be 16 characters or less. Given that FDB's own error handling here makes it hard to associate an error from option value validation with the origin, should this method validate the options (and throw if invalid)?

Reply:
Added the validation. I have no objection to `addTag`, if that seems cleaner. I just imagined that they all happened at once.

Reply:
I think either is fine, really.